xoinvader/ingame.py

```diff
@@ -88,6 +89,73 @@ def handle(self):
         # Some other event logic
 
 
+class TestWave(EnemyWave):
+    def __init__(self, state_object_adder, speed):
+        super(TestWave, self).__init__(speed)
+        self._state_object_adder = state_object_adder
+        self.add_event(1, self.spawn4)
+        self.add_event(100, lambda: None)
+
+    @staticmethod
+    def get_keyframes(y_offset, x_start, direction):
+        if direction == "left":
+            new_x = lambda d: x_start - d
+        elif direction == "right":
+            new_x = lambda d: x_start + d
+
+        return [
+            (0,   Point(x_start,   y_offset)),
+            (0.5, Point(x_start,   y_offset + 1)),
+            (1.0, Point(x_start,   y_offset + 2)),
+            (1.5, Point(x_start,   y_offset + 3)),
+            (2.0, Point(x_start,   y_offset + 4)),
+            (3.0, Point(new_x(1),  y_offset + 4)),
+            (3.5, Point(new_x(2),  y_offset + 5)),
+            (4.0, Point(new_x(4),  y_offset + 5)),
+            (4.3, Point(new_x(7),  y_offset + 6)),
+            (4.5, Point(new_x(9),  y_offset + 6)),
+            (4.7, Point(new_x(10), y_offset + 7)),
+            (4.7, Point(new_x(11), y_offset + 7)),
+            (4.7, Point(new_x(12), y_offset + 7)),
+            (4.9, Point(new_x(13), y_offset + 7)),
+            (5.0, Point(new_x(14), y_offset + 7)),
+            (5.1, Point(new_x(15), y_offset + 7)),
+            (5.2, Point(new_x(16), y_offset + 7)),
+            (5.3, Point(new_x(17), y_offset + 7)),
+            (5.4, Point(new_x(18), y_offset + 7)),
+            (5.5, Point(new_x(19), y_offset + 7)),
+            (5.6, Point(new_x(20), y_offset + 7)),
+            (5.7, Point(new_x(21), y_offset + 7)),
+            (5.8, Point(new_x(22), y_offset + 7)),
+        ]
+
+    def spawn4(self):
+        right_side = Settings.layout.field.edge.x
+
+        e1 = GenericXEnemy(Point(10, 1),
```

Comment:
here and below please format as I ask.
If function call line takes more than 80 (to be honest - I told 100 to pylint), format it like this:
```python
variable = long_function_name(
    param1, param2, param3, param4,
    another_long_function_name_call(
        arg1, arg2, arg3, long_long_long_long_arg,
        argN),
    another_arg)
```